packages/exp/src/commands/upload/BaseUploader.js

```diff
@@ -0,0 +1,224 @@
+import fs from 'fs';
+import process from 'process';
+
+import ProgressBar from 'progress';
+import Axios from 'axios';
+import _ from 'lodash';
+
+import { BuildInformation, ProjectUtils } from 'xdl';
+import * as UrlUtils from '../utils/url';
+import prompt from '../../prompt';
+import log from '../../log';
+
+const OPTIONS = ['path', 'latest', 'id'];
+
+export default class BaseUploader {
+  constructor(projectDir, options, platform, platformName, platformExtension) {
```

Comment:
Maybe create an object with `platform, platformName, platformExtension` arguments, it's better to have fewer arguments.